### PR TITLE
refactor: use RefCell for mutable state

### PR DESCRIPTION
**Pull request by bartlomieju**
Alternative to #3897

### CODE REVIEW

File: cli/Cargo.toml [hunk 1]
```diff
@@ -33,6 +33,7 @@ bytes = "0.5.3"
 byteorder = "1.3.2"
 brotli2 = "0.3.2"
 clap = "2.33.0"
+derive_deref = "1.1.0"
```
Comment (ry) [inline cli/Cargo.toml]: This seems unrelated to the goal of the patch.

I’d prefer if we continued to impl Deref manually - it’s not much work and don’t like adding dependencies.

Comment (piscisaureus) [reply cli/Cargo.toml]: I find myself writing trivial Deref/Derefmut impls so often that I recently just gave in.
